external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/CachedKafkaConsumer.scala

```diff
@@ -35,7 +35,7 @@ import org.apache.spark.sql.kafka010.KafkaSource._
  * Underlying consumer is not threadsafe, so neither is this,
  * but processing the same topicpartition and group id in multiple threads is usually bad anyway.
  */
-private[kafka010] case class CachedKafkaConsumer private(
+private[kafka010] case class CachedKafkaConsumer(
```

Comment:
Isnt it cleaner to create a new method `create` in object CachedKafkaConsumer which returns a new CachedKafkaConsumer without putting it in the map?